### PR TITLE
MINIFICPP-1502 Pass name and uuid arguments as const refs

### PR DESCRIPTION
**Pull request by lordgamez**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by szaszm — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

File: extensions/jni/JVMCreator.h [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace jni {
 class JVMCreator : public minifi::core::CoreComponent {
  public:
 
-  explicit JVMCreator(const std::string &name, utils::Identifier uuid = utils::Identifier())
+  explicit JVMCreator(const std::string &name, const utils::Identifier& uuid = {})
```
Comment (adamdebreceni) [inline extensions/jni/JVMCreator.h:43]: inconsistent `&` placing

File: extensions/opencv/CaptureRTSPFrame.h [hunk 2]
```diff
@@ -37,7 +37,7 @@ class CaptureRTSPFrame : public core::Processor {
 
  public:
 
-  explicit CaptureRTSPFrame(const std::string &name, utils::Identifier uuid = utils::Identifier())
+  explicit CaptureRTSPFrame(const std::string &name, const utils::Identifier& uuid = {})
```
Comment (adamdebreceni) [inline extensions/opencv/CaptureRTSPFrame.h:40]: here as well

File: extensions/opencv/MotionDetector.h [hunk 3]
```diff
@@ -40,7 +40,7 @@ class MotionDetector : public core::Processor {
 
  public:
 
-  explicit MotionDetector(const std::string &name, utils::Identifier uuid = utils::Identifier())
+  explicit MotionDetector(const std::string &name, const utils::Identifier& uuid = {})
```
Comment (adamdebreceni) [inline extensions/opencv/MotionDetector.h:43]: here as well

File: extensions/script/ExecuteScript.h [hunk 4]
```diff
@@ -36,7 +36,7 @@ namespace processors {
 
 class ExecuteScript : public core::Processor {
  public:
-  explicit ExecuteScript(const std::string &name, utils::Identifier uuid = utils::Identifier())
+  explicit ExecuteScript(const std::string &name, const utils::Identifier& uuid = {})
```
Comment (adamdebreceni) [inline extensions/script/ExecuteScript.h:39]: here as well

File: extensions/script/python/ExecutePythonProcessor.h [hunk 5]
```diff
@@ -45,7 +45,7 @@ namespace processors {
 
 class ExecutePythonProcessor : public core::Processor {
  public:
-  explicit ExecutePythonProcessor(const std::string &name, utils::Identifier uuid = utils::Identifier())
+  explicit ExecutePythonProcessor(const std::string &name, const utils::Identifier& uuid = {})
```
Comment (adamdebreceni) [inline extensions/script/python/ExecutePythonProcessor.h:48]: here as well

File: extensions/script/python/PythonCreator.h [hunk 6]
```diff
@@ -43,7 +43,7 @@ namespace python {
 class PythonCreator : public minifi::core::CoreComponent {
  public:
 
-  explicit PythonCreator(const std::string &name, utils::Identifier uuid = utils::Identifier())
+  explicit PythonCreator(const std::string &name, const utils::Identifier& uuid = {})
```
Comment (adamdebreceni) [inline extensions/script/python/PythonCreator.h:46]: here as well

File: extensions/jni/JVMCreator.h (hunk 1 above)
Comment (lordgamez) [reply extensions/jni/JVMCreator.h]: Fixed in 986bca3bf8ae4897cf73ce1fbe6530015f79989e

File: libminifi/include/core/Core.h [hunk 7]
```diff
@@ -148,7 +148,7 @@ class CoreComponent {
    * Constructor that sets the name and uuid.
    */
 
-  explicit CoreComponent(const std::string &name, const utils::Identifier& uuid = {}, const std::shared_ptr<utils::IdGenerator>& idGenerator = utils::IdGenerator::getIdGenerator())
+  explicit CoreComponent(const std::string &name, const utils::Identifier &uuid = {}, const std::shared_ptr<utils::IdGenerator>& idGenerator = utils::IdGenerator::getIdGenerator())
```
Comment (adamdebreceni) [inline libminifi/include/core/Core.h:151]: one more here

File: libminifi/include/RemoteProcessorGroupPort.h [hunk 8]
```diff
@@ -81,7 +81,7 @@ class RemoteProcessorGroupPort : public core::Processor {
   /*!
    * Create a new processor
    */
-  RemoteProcessorGroupPort(const std::shared_ptr<io::StreamFactory> &stream_factory, std::string name, std::string url, const std::shared_ptr<Configure> &configure, utils::Identifier uuid = utils::Identifier()) // NOLINT
+  RemoteProcessorGroupPort(const std::shared_ptr<io::StreamFactory> &stream_factory, const std::string& name, std::string url, const std::shared_ptr<Configure> &configure, const utils::Identifier& uuid = {}) // NOLINT
```
Comment (adamdebreceni) [inline libminifi/include/RemoteProcessorGroupPort.h:84]: one more inconsistency here

Comment (lordgamez) [reply libminifi/include/RemoteProcessorGroupPort.h]: Fixed in a8574d103f96566edec377a0d7e1447e33ef14b9

File: libminifi/include/core/Core.h (hunk 7 above)
Comment (lordgamez) [reply libminifi/include/core/Core.h]: Fixed in a8574d103f96566edec377a0d7e1447e33ef14b9

**Review by adamdebreceni — APPROVED** [review]

**Review by arpadboda — APPROVED** [review]

**Pull request closed without merging**